Add ability to assign trunks to vlans

nahun opened the pull request:
This adds trunks to the list of interfaces that can be assigned to a VLAN

caphrim007 commented:
@nahun I dont think `append` is what you want. I think `extend` is

nahun commented:
You're right, I didn't see that since its actually working for me. The error isn't being raised correctly it seems. If I define a non-existent interface its the BIG-IP that throws the error:
```
failed: [hostname -> localhost] (item={u'tag': 2223, u'name': u'test', u'tagged_interfaces': [u'1/1.17']}) => {"failed": true, "item": {"name": "test", "tag": 2223, "tagged_interfaces": ["1/1.17"]}, "msg": "400 Unexpected Error: Bad Request for uri: https://1.1.1.1:443/mgmt/tm/net/vlan/~Common~test/\nText: u'{\"code\":400,\"message\":\"01070277:3: The requested vlan member (1/1.17) was not found.\",\"errorStack\":[]}'"}
```
And then trying the _parse_return_ifcs method manually with the SDK throws an attribute error as I would expect, but the method doesn't fail when running a playbook.
```
>>> from f5.bigip import ManagementRoot
>>> mgmt = ManagementRoot("1.1.1.1", "admin", "admin")
>>> l = mgmt.tm.net.interfaces.get_collection()
>>> l.append(mgmt.tm.net.trunks.get_collection())
>>> t = [str(x.name) for x in l]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list' object has no attribute 'name'
```

caphrim007 commented:
the bigger question is why didn't you get an exception using the trunks API.

because there is no trunks API

caphrim007 commented:
what version of python are you using

nahun commented:
Oh man I completely forgot I added that myself as well. Alright next time I won't put in a pull request months after I do some work.

In the F5 SDK I added a simple bigip/tm/net/trunk.py that is pretty much copied from interface.py. Maybe I'll contribute it to f5-common-python.

Closing this since there isn't even support for it in the SDK. Sorry about that.

caphrim007 commented:
well, I added and pushed the vcmp addition you made. so that should now be in the provision code

nahun commented:
ok thanks, I also didn't mean for that to be in this request. I was going to open a different one. I of course should have created different branches for those...
